File: docs/entities/entityLogs.doc.cs
using System;

public enum ActionType
{
    CREATE,
    UPDATE,
    DELETE
}

public class EntityLogs
{
    public string Service { get; set; }
    public string EntityName { get; set; }
    public ActionType Action { get; set; }
    public DateTime Timestamp { get; set; }
    public string? UserId { get; set; } = null;
    public string? UserName { get; set; } = null;
    public dynamic? Changes { get; set; } = null;
    public dynamic? OldValues { get; set; } = null;
    public dynamic? NewValues { get; set; } = null;
}

File: docs/entities/entityLogs.doc.ts
export interface EntityLogs {
    service: string;
    entityName: string;
    action: ActionType;
    timestamp:Date;
    userId?: string;
    userName?: string;
    changes?: any;
    oldValue?:any;
    newValue?:any;
}

export enum ActionType {
    CREATE = 'CREATE',
    UPDATE = 'UPDATE',
    DELETE = 'DELETE'
}